src/storage/host_store.rs

```diff
@@ -127,15 +127,38 @@ impl PageStorage {
     }
 
     fn get(&self, ofs: &OffsetLen) -> &[u8] {
+        fn page_for_ofs(pages_ofs: usize, pages: &Vec<Page>) -> usize {
```

Comment:
I think this could be simplified to not have `pages` now, not certain the extra functions are needed.